src/main/kotlin/com/jetbrains/snakecharm/codeInsight/SnakemakeParamsInShellReferenceContributor.kt

```diff
@@ -0,0 +1,48 @@
+package com.jetbrains.snakecharm.codeInsight
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.*
+import com.intellij.psi.util.parentOfType
+import com.intellij.util.ProcessingContext
+import com.jetbrains.python.psi.PyStringLiteralExpression
+import com.jetbrains.snakecharm.lang.SnakemakeLanguageDialect
+import com.jetbrains.snakecharm.lang.psi.SMKParamsReference
+import com.jetbrains.snakecharm.lang.psi.SMKRuleParameterListStatement
+import java.util.regex.Pattern
+
+class SnakemakeParamsInShellReferenceContributor : PsiReferenceContributor() {
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        registrar.registerReferenceProvider(
+                PlatformPatterns.psiElement(),
+                object : PsiReferenceProvider() {
+                    private val paramsPattern = Pattern.compile("\\{params\\.(.+?)}")
```

Comment:
Do we want to accept any suffix after `params.` e.g. `foo.boo` in `params.foo.boo`? Seems we expect only identifier after dot.

Reply:
I don't quite get what's the desired behavior here, probably because I've been unable to find an example of nested parameters declaration syntax (how would I go about declaring `foo.boo` in the `params` section?). As of now, any attempt at writing `params.foo.boo` when only `foo` is defined results in an unresolved reference error. What should the user get instead? I could modify the regex to not match `params.foo.`, then there would be no error, but I'm not sure if that's what's expected here. I don't really know what happens in case if `params.foo.boo` should actually get resolved because I'm unable to declare such a construction.

Reply:
I think, that we cannot process nested params at the moment and thus I expect to match only one identifier after `params.`, not any complex expression.